From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: forbiddenlake <2653277+chennin@users.noreply.github.com>
Date: Sat, 15 Mar 2025 15:56:47 -0400
Subject: [PATCH] docs: Fix HTML build with Sphinx 8.2

Sphinx 8.2 changed the definition of `depart_admonition`, causing build
failures when building with html-docs=enabled as mesa was only
overloading `visit_admonition`. This adds the old `depart_admonition`
definition to mesa's copy, restoring compatibility and fixing the build.

Closes: https://gitlab.freedesktop.org/mesa/mesa/-/issues/12725
Part-of: <https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/34165>
---
 docs/_exts/bootstrap.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/docs/_exts/bootstrap.py b/docs/_exts/bootstrap.py
index 18e46cea1608..c00042452018 100644
--- a/docs/_exts/bootstrap.py
+++ b/docs/_exts/bootstrap.py
@@ -71,6 +71,9 @@ class BootstrapHTML5TranslatorMixin:
             self.body.append(str(admonitionlabels[name]))
             self.body.append('</div>')
 
+    def depart_admonition(self, node) -> None:
+        self.body.append('</div>\n')
+
     def visit_table(self, node):
         # init the attributes
         atts = {}
